### PR TITLE
tokio: make all windows docs visible in unix builds

### LINKED ISSUE #5499: missing documentation for Windows-specific tokio::fs functions
**Version**

`1.25.0`

**Platform**

Windows

**Description**

While working on PR #5493 I discovered that documentation for Windows-specific functions is missing. For example:

- `tokio::fs::OpenOptions` functions:
  - `custom_flags`: only Unix version documented here: https://docs.rs/tokio/1.25.0/tokio/fs/struct.OpenOptions.html#method.custom_flags
  - `access_mode`, `share_mode`, `attributes`, `security_qos_flags`
- top-level `tokio-fs` functions - https://docs.rs/tokio/1.25.0/tokio/fs/index.html#functions - has Unix `symlink` function but missing Windows-specific `symlink_dir` & `symlink_file` functions

I did find the Windows-specific functions documented for an earlier version of Tokio:

- https://docs.rs/tokio/1.7.0/i686-pc-windows-msvc/tokio/fs/struct.OpenOptions.html#method.access_mode etc.
- https://docs.rs/tokio/1.7.0/i686-pc-windows-msvc/tokio/fs/fn.symlink_dir.html
- https://docs.rs/tokio/1.7.0/i686-pc-windows-msvc/tokio/fs/fn.symlink_file.html

but when I tried the platform dropdown for Tokio 1.25.0 I did not see an option for Windows.

Please let me know if I am missing anything. I hope to investigate this further (someday). Thanks.

### PR DESCRIPTION
**Pull request by satakuma**
This PR makes all Windows-specific documentation visible in Unix builds similarly to the docs for named pipes.

Items which should be now visible:
- `tokio::fs::symlink_dir`
- `tokio::fs::symlink_file`
- `tokio::fs::OpenOptions` methods (custom_flags, access_mode, share_mode, attributes, security_qos_flags)
- `{AsRawHandle, FromRawHandle}` impls for `tokio::fs::File`
- `AsRawHandle` impl for `tokio::io::{Stdin, Stdout, Stderr}`
- `AsRawSocket` impl for `tokio::net::{TcpListener, TcpStream, UdpSocket}`
- `{AsRawSocket, FromRawSocket, IntoRawSocket}` for `tokio::net::TcpSocket`
- `AsRawHandle` impl for `tokio::process::{ChildStdin, ChildStdout, ChildStderr}`
- `tokio::process::Command::creation_flags`
- `tokio::process::Child::raw_handle`

[View rendered docs](https://deploy-preview-5530--tokio-rs.netlify.app/tokio/)

Resolves #5499

### CODE REVIEW

**Comment by satakuma** [comment]
Now I see that this will merge conflict with #5514, I can wait until that PR is merged first.

**Comment by satakuma** [comment]
Updated to include `AsHandle`/`AsSocket` implementations as well.

**Review by Darksonn — APPROVED** [review]

**Pull request merged**
